File: src/main/java/ru/nsu/ccfit/dmakogon/storage/ReplicatedHashMap.java
package ru.nsu.ccfit.dmakogon.storage;

import java.util.List;

public interface ReplicatedHashMap<K, V> {
  V get(K key);
  void insert(K key, V value);
  void update(K key, V value);
  void remove(K key);

  List<ReplicatedHashMapEntry<K, V>> all();
}

File: sync-hashmap-python/replicated_hashmap.py
class ReplicatedHashMap():
    def __init__(self):
        self.map = {}
    
    def get(self, key):
        return self.map[key]

    def insert(self, key, value):
        self.map[key] = value

    def update(self, key, value):
        self.map[key] = value

    def remove(self, key):
        del self.map[key]

    def all(self):
        return self.map.items()
